numpyro/infer/util.py

```diff
@@ -673,17 +674,47 @@ def _predictive(
     posterior_samples,
     batch_shape,
     return_sites=None,
+    infer_discrete=False,
+    temperature=1,
```

Comment:
nit: maybe call this `infer_discrete_temperature` to distinguish from sampling from annealed posteriors?

Reply:
Good idea! I'll make the change.